param/__init__.py

```diff
@@ -1275,6 +1445,19 @@ def __init__(self, objects=None, default=None, instantiate=False,
         if default is not None and self.check_on_set is True:
             self._validate(default)
 
+    @property
+    def objects(self):
+        return ListProxy(self._objects, self)
+
+    @objects.setter
+    def objects(self, objects):
+        if isinstance(objects, collections_abc.Mapping):
+            self.names = objects
+            self._objects = list(objects.values())
+        else:
+            self.names = {}
```

Comment:
The default value of `names` is now an empty dict when `objects` is a list, instead of `None`. The test suite fails now, since I added a bunch of tests that check the default values.

Seems like this change was intentional, @philippjfr can you confirm? If so, I can update the tests accordingly.

Reply:
I assumed this was an intentional change and updated the tests accordingly. Can always amend before 2.0 is release if need be.